### ISSUE TITLE
cfn_clean breaks DefinitionString with Sub function usage

### ISSUE THREAD

**Issue by lanwen** [cloudtools/troposphere#1423]
It produces yaml in form:
```yaml
      DefinitionString: |-
        {
          "Fn::Sub": [
            "\n        {\n           \"QueueUrl\": \"${feedbackQueueUrl}\"\n   ...    }\n        ",
            {
              "feedbackQueueUrl": {
                "Ref": "FeedbackRequestsQueue"
              }
            }
          ]
        }
```

when should be 

```yaml
      DefinitionString: !Sub
        - "\n         \"QueueUrl\": \"${feedbackQueueUrl}\"\n                    },\n\
          \                    \"End\": true\n                }\n            }\n \
          \       }\n        "
        - feedbackQueueUrl: !Ref 'FeedbackRequestsQueue'
```

the full python scrypt to reproduce that:

```python
#!/usr/bin/env python2

from troposphere import Template, GetAtt, Ref, Sub, Parameter, Split, Join, AWSObject
from troposphere import ecs, cloudformation, iam, stepfunctions, sqs

t = Template()

feedback_requests_queue = t.add_resource(sqs.Queue(
    'FeedbackRequestsQueue',
    VisibilityTimeout=60 * 10,
    MessageRetentionPeriod=60 * 60 * 24 * 2
))

feedback_request_state_machine_role = t.add_resource(iam.Role(
    'FeedbackRequestStateMachineRole',
    AssumeRolePolicyDocument={
        "Statement": [
            {"Effect": "Allow", "Principal": {"Service": [Sub("states.${AWS::Region}.amazonaws.com")]}, "Action": ["sts:AssumeRole"]}]
    },
    Policies=[
        iam.PolicyProperty(
            PolicyName='submit_request_to_sqs',
            PolicyDocument={
                "Statement": [
                    {"Effect": "Allow", "Action": ["sqs:SendMessage"], "Resource": GetAtt(feedback_requests_queue, "Arn")},
                ],
            },
        ),
    ]
))

feedback_request_state_machine = t.add_resource(stepfunctions.StateMachine(
    'FeedbackRequestStateMachine',
    StateMachineName=Sub('${AWS::StackName}-FeedbackRequestStateMachine'),
    DefinitionString=Sub(
        '''
        {
            "Comment": "State machine which awaits until defined date to submit a feedback request",
            "StartAt": "WaitForDueDate",
            "States": {
                "WaitForDueDate": {
                    "Type": "Wait",
                    "TimestampPath": "$.plannedAt",
                    "Next": "SubmitFeedbackRequestToSQS"
                },
                "SubmitFeedbackRequestToSQS": {
                    "Type": "Task",
                    "Resource": "arn:aws:states:::sqs:sendMessage",
                    "Parameters": {
                        "MessageBody.$": "$",
                        "QueueUrl": "${feedbackQueueUrl}"
                    },
                    "End": true
                }
            }
        }
        ''',
        feedbackQueueUrl=feedback_requests_queue.ref()
    ),
    RoleArn=Join("/", [Sub("arn:aws:iam::${AWS::AccountId}:role"), feedback_request_state_machine_role.ref()]),
))


print(t.to_yaml())
```

it seems that with `to_json()` it doesn't have any issues

any workarounds from troposphere side?

**Comment by markpeek** [comment]
This looks to be an issue/regression in cfn_flip v1.2.1. Try installing this: `pip install -U 'cfn_flip==1.2.0'` and see if it works for you.

**Comment by lanwen** [comment]
Yes, that fixes the issue

**Comment by sblask** [comment]
I think the resulting string should really be
```
            DefinitionString: !Sub |
                {
                    "Comment": "State machine which awaits until defined date to submit a feedback request",
                    "StartAt": "WaitForDueDate",
                    "States": {
                        "WaitForDueDate": {
                            "Type": "Wait",
                            "TimestampPath": "$.plannedAt",
                            "Next": "SubmitFeedbackRequestToSQS"
                        },
                        "SubmitFeedbackRequestToSQS": {
                            "Type": "Task",
                            "Resource": "arn:aws:states:::sqs:sendMessage",
                            "Parameters": {
                                "MessageBody.$": "$",
                                "QueueUrl": "${FeedbackRequestsQueue}"
                            },
                            "End": true
                        }
                    }
                }
```
I just tried to switch to yaml for my step function written using troposphere to get a nicely formatted definition string instead of one line with loads of `\n`. I was really surprised that Sub works as expected for everything but step function definition :-( After some research, this might be related to https://github.com/awslabs/aws-cfn-template-flip/pull/80 but it would be really great if this could be fixed!

**Comment by stilvoid** [comment]
I've just released cfn-flip 1.2.2 which contains this bugfix.
